File: Seori/Weekly Contest 401/3178. Find the Child Who Has the Ball After K Seconds.py
class Solution:
    def numberOfChild(self, n: int, k: int) -> int:
        # 왕복하는데 걸리는 시간 2*(n-1)이므로 k를 마지막 사이클만 남김
        net_k = k % (2 * (n-1))

        # 마지막 사이클에서 quotient가 0이면 정방향, 1이면 역방향
        quotient = net_k // (n-1)
        remainder = net_k % (n-1)

        return remainder if not quotient else n - 1 - remainder

File: Gombang/Weekly Contest 401/3178_Find_the_Child_Who_Has_the_Ball_After_K_Seconds.cs
public class Solution
{
    public int NumberOfChild(int n, int k)
    {
        const int RIGHT = 1;
        const int LEFT = -1;

        int current = 0;
        int direction = RIGHT;

        for (int i = 0; i < k; i++)
        {
            current += direction;

            if (current == n - 1)
                direction = LEFT;
            else if (current == 0)
                direction = RIGHT;
        }

        return current;
    }
}